### PR TITLE
Tor blutmagie

### PR DESCRIPTION
**Pull request by srilumpa**
As for PR #138, this PR aims to offer a solution to issue #45 with an analyzer extracting data from torstatus.blutmagie.de to check if an artifact is linked to a TOR node. The artifact can be an IP address, a FQDN or a domain.

I documented both of the classes but if you need a more advance documentation, please tell me. Also, I hope i coped with your coding standards. If not tell me so I can fix it.

As for #138, there is a caching system in place.

### CODE REVIEW

**Comment by 3c7** [comment]
Thanks for the fast response. Caching works great.

- [x] Python3 compatibility
- [x] Caching
- [x] Shell test
```
(TorBlutmagie) ➜  TorBlutmagie git:(TorBlutmagie) ✗ ./tor_blutmagie_analyzer.py <<< '{
"dataType": "ip",
"data": "134.119.3.2"
}' | jq
{
  "success": true,
  "summary": {
    "taxonomies": [
      {
        "level": "suspicious",
        "namespace": "TorBlutmagie",
        "predicate": "Node",
        "value": "1 node"
      }
    ]
  },
  "artifacts": [
    {
      "type": "fqdn",
      "value": "j72505.servers.jiffybox.net"
    },
    {
      "type": "ip",
      "value": "134.119.3.2"
    }
  ],
  "full": {
    "nodes": [
      {
        "hostname": "j72505.servers.jiffybox.net",
        "name": "Zwiebel2",
        "country_code": "DE",
        "ip": "134.119.3.2",
        "as_name": "HOSTEUROPE-AS- DE",
        "as_number": "20773"
      }
    ]
  }
}
(TorBlutmagie) ➜  TorBlutmagie git:(TorBlutmagie) ✗ ./tor_blutmagie_analyzer.py <<< '{
"dataType": "fqdn",
"data": "j72505.servers.jiffybox.net"
}' | jq
{
  "success": true,
  "summary": {
    "taxonomies": [
      {
        "level": "suspicious",
        "namespace": "TorBlutmagie",
        "predicate": "Node",
        "value": "1 node"
      }
    ]
  },
  "artifacts": [
    {
      "type": "fqdn",
      "value": "j72505.servers.jiffybox.net"
    },
    {
      "type": "ip",
      "value": "134.119.3.2"
    }
  ],
  "full": {
    "nodes": [
      {
        "hostname": "j72505.servers.jiffybox.net",
        "name": "Zwiebel2",
        "country_code": "DE",
        "ip": "134.119.3.2",
        "as_name": "HOSTEUROPE-AS- DE",
        "as_number": "20773"
      }
    ]
  }
}
(TorBlutmagie) ➜  TorBlutmagie git:(TorBlutmagie) ✗ ./tor_blutmagie_analyzer.py <<< '{
"dataType": "domain",
"data": "jiffybox.net" 
}' | jq
{
  "success": true,
  "summary": {
    "taxonomies": [
      {
        "level": "suspicious",
        "namespace": "TorBlutmagie",
        "predicate": "Node",
        "value": "7 nodes"
      }
    ]
  },
  "artifacts": [
    {
      "type": "fqdn",
      "value": "j98727.servers.jiffybox.net"
    },
    {
      "type": "ip",
      "value": "93.180.156.84"
    },
    {
      "type": "fqdn",
      "value": "j60204.servers.jiffybox.net"
    },
    {
      "type": "ip",
      "value": "37.200.99.251"
    },
    {
      "type": "fqdn",
      "value": "j17531.servers.jiffybox.net"
    },
    {
      "type": "ip",
      "value": "46.252.26.2"
    },
    {
      "type": "fqdn",
      "value": "j54490.servers.jiffybox.net"
    },
    {
      "type": "ip",
      "value": "37.200.98.5"
    },
    {
      "type": "fqdn",
      "value": "j184462.servers.jiffybox.net"
    },
    {
      "type": "ip",
      "value": "134.119.36.135"
    },
    {
      "type": "fqdn",
      "value": "j32098.servers.jiffybox.net"
    },
    {
      "type": "ip",
      "value": "37.200.98.117"
    },
    {
      "type": "fqdn",
      "value": "j72505.servers.jiffybox.net"
    },
    {
      "type": "ip",
      "value": "134.119.3.2"
    }
  ],
  "full": {
    "nodes": [
      {
        "hostname": "j98727.servers.jiffybox.net",
        "name": "BARACUDA",
        "country_code": "DE",
        "ip": "93.180.156.84",
        "as_name": "HOSTEUROPE-AS- DE",
        "as_number": "20773"
      },
      {
        "hostname": "j60204.servers.jiffybox.net",
        "name": "JPsi2",
        "country_code": "DE",
        "ip": "37.200.99.251",
        "as_name": "HOSTEUROPE-AS- DE",
        "as_number": "20773"
      },
      {
        "hostname": "j17531.servers.jiffybox.net",
        "name": "marlen1",
        "country_code": "DE",
        "ip": "46.252.26.2",
        "as_name": "HOSTEUROPE-AS- DE",
        "as_number": "20773"
      },
      {
        "hostname": "j54490.servers.jiffybox.net",
        "name": "torpidsDEdomainf",
        "country_code": "DE",
        "ip": "37.200.98.5",
        "as_name": "HOSTEUROPE-AS- DE",
        "as_number": "20773"
      },
      {
        "hostname": "j184462.servers.jiffybox.net",
        "name": "torpidsDEdomainf2",
        "country_code": "DE",
        "ip": "134.119.36.135",
        "as_name": "HOSTEUROPE-AS- DE",
        "as_number": "20773"
      },
      {
        "hostname": "j32098.servers.jiffybox.net",
        "name": "Unnamed",
        "country_code": "DE",
        "ip": "37.200.98.117",
        "as_name": "HOSTEUROPE-AS- DE",
        "as_number": "20773"
      },
      {
        "hostname": "j72505.servers.jiffybox.net",
        "name": "Zwiebel2",
        "country_code": "DE",
        "ip": "134.119.3.2",
        "as_name": "HOSTEUROPE-AS- DE",
        "as_number": "20773"
      }
    ]
  }
}

```
- [ ] TheHive Integration Test (Reports etc.)

File: thehive-templates/TorBlutmagie_1_0/long.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	<div class="panel-heading">
+		Tor nodes - {{artifact.data | fang}}
+	</div>
+	<div class="panel-body">
+		<div ng-repeat="n in content.nodes" class="panel panel-default">
+      <div class="panel-heading">
+        {{n.name}}
+      </div>
+      <div class="panel-body">
+        <dl class="dl-horizontal">
```
Comment (3c7) [inline thehive-templates/TorBlutmagie_1_0/long.html]: Please switch the `<dt>` and `<dd>` here:

grafik

File: thehive-templates/TorBlutmagie_1_0/long.html [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          <dd>Address</dd>
+          <dt>{{n.hostname | fang}} ({{n.ip_address | fang}})</dt>
```
Comment (3c7) [inline thehive-templates/TorBlutmagie_1_0/long.html]: Here, it has to be `n.ip`.

File: thehive-templates/TorBlutmagie_1_0/long.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          <dd>AS</dd>
+          <dt>{{n.as_name}} ({{n.as_number}})</dt>
+          <dd>Country</dd>
+          <dt>{{n.country_code}}</dt>
+        </dl>
+        <span>
+          <i class="fa fa-search"></i>
+          <a ng-href="http://torstatus.blutmagie.de/cgi-bin/whois.pl?ip={{n.ip_address}}" target="_blank">WHOIS</a>
```
Comment (3c7) [inline thehive-templates/TorBlutmagie_1_0/long.html]: This has to be `n.ip`, also.

**Comment by srilumpa** [comment]
Here are the fixes. I am so sorry for those basic errors.

**Pull request merged**